core/src/main/java/feign/Response.java

```diff
@@ -92,6 +91,11 @@ public String reason() {
     return reason;
   }
 
+  /**
+   * Map of header names to collection of values. This is a TreeMap which uses a case-insensitive
```

Comment:
no need to talk about implementation (in case it one day changes), you can just say..

```
/** Returns a case-insensitive mapping of header names to their values. */
```